services/getifaddrs.cpp

```diff
@@ -212,7 +212,12 @@ int kde_getifaddrs(struct kde_ifaddrs **ifap)
                 break;
             }
 
+#ifdef __sgi
+            struct sockaddr_in* sain = (struct sockaddr_in*) &ifr->ifr_addr;
+            storage[i].netmask = *(struct sockaddr *)sain->sin_addr.s_addr;
```

Comment:
This is what I segfault on, but have hardcoded it and gotten the daemon to start